lib/webmachine.mli

```diff
@@ -312,5 +312,5 @@ module type S = sig
         {[([`Lit, "user"; `Var "id"], `Prefix, user_resource)]} *)
 end
 
-module Make(IO:IO) : S
+module Make(IO:IO)(DATE:Wm_util.Date_sig) : S
```

Comment:
I'm pretty sure `Wm_util` isn't an exposed module, so the signature should be defined in this file and then just used below. That's what's done for `IO` as well, at the top of the file.

Reply:
Not sure what you mean. It _does_ work, if that's what you doubt. Or is that a style question?

Reply:
`Webmachine.Wm_util.Date_sig` would not be available to users of the library. So anybody that wanted to use that signature wouldn't be able to get a handle on it. Also, it's not clear what that signature _is_ from the `.mli`, so that folks can easily implement a module that satisfies it.

Reply:
Wm_util doesn't have an mli file, so in my understanding it _is_ available. Or do other restriction apply here?

Here I ported irmin-http to the new webmachine, and it works: https://github.com/ansiwen/irmin/commit/e1af324883d393928755647bd07f59e07715284e 

But nevertheless, I agree that it's not beautiful. I would also prefer to hide Wm_util from the outside world. But then the whole functor has to go into webmachine.mli, if I'm not mistaken?

Reply:
Got it, I thought that `Wm_util` wasn't exposed but it is. That's an error and should be fixed in a separate issue (not your problem but I suppose relevant to this feature). Ideally, only the signature would be available in the `.mli` and there would be no need for a functor. (Providing functors to create modules to pass to functors is a recipe for a headache). One way to achieve that is to make the type in the signature concrete. In my comment below I essentially suggested changing the time of now to `unit -> int` where the return value is seconds since UNIX epoch. That would achieve it, and also lead to a simple implementation for UNIX backends:

```ocaml
let now () = int_of_float (Unix.gettimeofday ())
```